plugins/plugins/sentinel/config.go

```diff
@@ -0,0 +1,105 @@
+// Copyright The HTNN Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sentinel
+
+import (
+	sentinelApi "github.com/alibaba/sentinel-golang/api"
+	sentinelConf "github.com/alibaba/sentinel-golang/core/config"
+	"github.com/alibaba/sentinel-golang/logging"
+
+	"mosn.io/htnn/api/pkg/filtermanager/api"
+	"mosn.io/htnn/api/pkg/plugins"
+	"mosn.io/htnn/plugins/plugins/sentinel/rules"
+	"mosn.io/htnn/types/plugins/sentinel"
+)
+
+func init() {
+	plugins.RegisterPlugin(sentinel.Name, &plugin{})
+}
+
+type plugin struct {
+	sentinel.Plugin
+}
+
+func (p *plugin) Factory() api.FilterFactory {
+	return factory
+}
+
+func (p *plugin) Config() api.PluginConfig {
+	return &config{}
+}
+
+type config struct {
+	sentinel.CustomConfig
+
+	params      sentinelApi.EntryOption
+	attachments []*sentinel.Source
+	m           *res2RuleMap
+}
+
+type res2RuleMap struct {
+	f  map[string]*sentinel.FlowRule
+	hs map[string]*sentinel.HotSpotRule
+	cb map[string]*sentinel.CircuitBreakerRule
+}
+
+func (conf *config) Init(cb api.ConfigCallbackHandler) error {
+	sc := sentinelConf.NewDefaultConfig()
+	sc.Sentinel.Log.Logger = logging.NewConsoleLogger()
+	sc.Sentinel.Log.Dir = "/tmp/sentinel"
+	if err := sentinelApi.InitWithConfig(sc); err != nil {
+		return err
+	}
+
+	if _, err := loadRules(conf); err != nil {
```

Comment:
Because `ok` only represents whether the rules loaded this time are different from the last time, if there is a difference, it is `true`, otherwise it is `false`. As far as I know, plugins don't trigger multiple loads. I'll simplify it

Reply:
I see. Would you change the `ok` to a better var name?

Reply:
I think I should get rid of the return value `ok`, because I will ignore it in the child function, that is, I ignore whether it loads different rules.